SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialReceiver.cpp

```diff
@@ -840,17 +848,70 @@ void USpatialReceiver::OnComponentUpdate(Worker_ComponentUpdateOp& Op)
 	}
 	else if (Category == ESchemaComponentType::SCHEMA_NetMulticastRPC)
 	{
-		if (const TArray<UFunction*>* RPCArray = Info.RPCs.Find(SCHEMA_NetMulticastRPC))
-		{
-			ReceiveMulticastUpdate(Op.update, TargetObject, *RPCArray);
-		}
+		checkNoEntry();
 	}
 	else
 	{
 		UE_LOG(LogSpatialReceiver, Verbose, TEXT("Entity: %d Component: %d - Skipping because it's an empty component update from an RPC component. (most likely as a result of gaining authority)"), Op.entity_id, Op.update.component_id);
 	}
 }
 
+void USpatialReceiver::HandleUnreliableRPC(Worker_ComponentUpdateOp& Op)
+{
+	Worker_EntityId EntityId = Op.entity_id;
+	if (StaticComponentView->HasAuthority(EntityId, Op.update.component_id))
+	{
+		// If we have authority over the component then we sent the event, so ignore.
+		return;
+	}
+
+	switch (Op.update.component_id)
+	{
+		case SpatialConstants::CLIENT_RPCS_COMPONENT_ID:
+		{
+			if (!StaticComponentView->HasAuthority(EntityId, SpatialConstants::SERVER_RPCS_COMPONENT_ID))
+			{
+				// We do not have We are not the server this RPC is looking for
+				return;
+			}
+			break;
+		}
+		case SpatialConstants::SERVER_RPCS_COMPONENT_ID:
+		{
+			if (!StaticComponentView->HasAuthority(EntityId, SpatialConstants::CLIENT_RPCS_COMPONENT_ID))
+			{
+				// We are not the client this RPC is looking for.
+				return;
+			}
+		}
+		break;
+		default:
+		break;
+	}
+	Schema_Object* EventsObject = Schema_GetComponentUpdateEvents(Op.update.schema_type);
+
+	uint32 EventCount = Schema_GetObjectCount(EventsObject, 1);
```

Comment:
Like we do in the GlobalStateManager, could we make constants for these field ids rather than having a naked 1.